jadx-core/src/main/java/jadx/api/ResourceFile.java

```diff
@@ -53,22 +56,48 @@ public ResContainer loadContent() {
 		return ResourcesLoader.loadContent(decompiler, this);
 	}
 
-	public boolean setAlias(ResourceEntry ri) {
+	public boolean setAlias(ResourceEntry entry, boolean useHeders) {
 		StringBuilder sb = new StringBuilder();
-		sb.append("res/").append(ri.getTypeName()).append(ri.getConfig());
-		sb.append("/").append(ri.getKeyName());
-		int lastDot = name.lastIndexOf('.');
-		if (lastDot != -1) {
-			sb.append(name.substring(lastDot));
+		sb.append("res/").append(entry.getTypeName()).append(entry.getConfig());
+		sb.append("/").append(entry.getKeyName());
+
+		if (useHeders) {
+			try {
+				byte[] bytes = ResourcesLoader.decodeStream(this, (size, is) -> is.readAllBytes());
```

Comment:
Considering that resources can get pretty large I would recommend not to perform file detection on the full content. As far as I know most file type detectors limit their detection on the start of the file, e.g. the first 4K.